src/Common/Config/ConfigProcessor.cpp

```diff
@@ -931,9 +961,34 @@ void ConfigProcessor::savePreprocessedConfig(LoadedConfig & loaded_config, std::
     }
 
 #if USE_SSL
-    std::string preprocessed_file_name = fs::path(preprocessed_path).filename();
-    if (preprocessed_file_name == "config.xml" || preprocessed_file_name == std::format("config{}.xml", PREPROCESSED_SUFFIX))
+    /* If skip_zk_encryption_keys == true, load encryption codecs and decrypt config's encrypted elements unless encryption keys are stored in ZK,
```

Comment:
I am looking at the condition in l. 987. I think we should slightly re-phrase l. 964-965:

```
Some callers (e.g. during very early server startup) will not have access to Zookeeper. Such callers can specify `skip_zk_encryption_keys = false` (but they will not be able to decrypt encrypted elements). If there are no `encryption_codecs` tags with `from_zk` attributes, we can decrypt anyways.

Config example we process here:
[...]
```